documents/Specification-Format-Style-Guidelines.md

```diff
@@ -0,0 +1,296 @@
+     description or requirements, not here.
+      - A good rule of thumb is that if this first sentence exceeds 60
+        characters, then it is trying to describe too much.
+- Include a line break (`<br/>`) after the first line, then a brief description,
+  if appropriate.
+   - Not every field needs additional description, but _every_ `message` does.
+   - Descriptions may be detailed, but should not include internal
+     implementation details or requirements. The description _should_ be
+     detailed enough to explain to an external contributor what the
+     message or field contains, where it fits within the overall network
+     system, and why it is needed.
+      - This is a long-term goal; we have a long way to go to get to this
+        level of description. Every PR to change `.proto` files should seek
+        to improve the documentation for that file.
+   - Avoid describing what must, should, must not, or should not be
+     sent, or how the field values affect output. Those items belong in the
+     _requirements_, instead.
+- After a paragraph break (a blank line for messages), one requirement per
+  line. Use break tags at the end of each requirement except the last.
+   - In field comments a `<p>` tag is _required_ for paragraph breaks.
+     Blank lines break the automated documentation generator in **fields**
+     because fields are output as markdown tables, and markdown tables require
+     each row to be a single long line.
+- Be _prescriptive_ in requirements.  Do not _describe_ the behavior.
+   - Declare what MUST be done for valid inputs, and what SHALL be present in
+     valid outputs (or in state messages).
+   - Declare what SHALL happen as a result of a successful, or failed,
+     transaction.
+   - The word `will` usually means `SHALL` or `MUST`.
+   - The word `is` often means `SHALL be`.
+   - Document both positive (`SHALL` or `MUST`) and negative (`SHALL NOT` or
+     `MUST NOT`) requirements.
+   - Avoid "perfect" or "imperfect" verb aspects and "wiggle" words
+     (might, could, perhaps, etc...).
+      - The specification items MAY, SHOULD, and RECOMMENDED serve to fill
+        the _occasional_ need for less imperative requirements.
+   - Also avoid "progressive" verbs (e.g. working, writing, going, deleting).
+- Requirements should be short and focused.
+   - Better to have three separate lines for three items than one longer
+     line with three clauses.
+   - Duplicating text across lines with slightly different requirements
+     is not only acceptable, but often recommended.
+   - Well written specification tends to be much more short and clipped
+     sentences than "normal" prose. Some describe the flow as "staccato".
+- Add another blank line after requirements for `message` documents only.
+- Each message that represents a transaction body **must** document the
+  Block Stream effects with a heading `### Block Stream Effects`.
```

Comment:
Could also add here that any heading such as `### Block Stream Effects` should not have a `<br/> `since titles are block elements (like paragraphs)

Reply:
Updated.